compositor_pipeline/src/pipeline/decoder/fdk_aac.rs

```diff
@@ -88,6 +64,30 @@ fn run_decoder_thread(
                 break;
             }
         };
+
+        let decoder = match &decoder {
+            Some(d) => d,
+            None => {
+                decoder = match Decoder::new(
+                    std::mem::take(&mut options)
+                        .expect("AAC decoder initialization options should be present"),
+                    &chunk,
```

Comment:
refactor this, initializing decoder this way is very hacky

Reply:
i refactored it a bit, see if it's better now

Reply:
The main problem with that code is that you are initializing that in the loop.